### PR TITLE
Exercise 9

### PR DESCRIPTION
**Pull request by emifly**

### CODE REVIEW

File: .github/workflows/ci-cd-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        uses: actions/checkout@v3
+      - name: Set Docker Hub username environment variable
+        run: echo "${{ secrets.ENV_FILE }}" | grep "^DOCKERHUB_USERNAME=" >> $GITHUB_ENV
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ env.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Set environment variables
+        run: echo "${{ secrets.ENV_FILE }}" > .env
```
Comment (JackMead) [inline .github/workflows/ci-cd-pipeline.yml:59]: Do you actually need this .env here? It's not used to build your images (I hope!) so I _think_ we could drop it here

**Review by JackMead — APPROVED** [review]
Nice work, you've got a thorough & well designed pipeline, and I can see it's happily pushing your latest Docker image 😄 I do think it's not quite working for the webhook request, but see my comment below for that. Once that's working, more than happy for this to be merged

File: .github/workflows/ci-cd-pipeline.yml (hunk 1 above)
Comment (emifly) [reply .github/workflows/ci-cd-pipeline.yml]: It totally was being used to build the images, have attempted to change that now!

File: .github/workflows/ci-cd-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        run: docker compose build production
+      - name: Push Docker image
+        run: docker compose push production
+  azure:
+    name: Refresh Azure app to pull from latest available Docker image
+    needs: docker
+    runs-on: ubuntu-latest
+    steps:
+      - name: Make post request to Azure app webhook URL
+        run: curl -dH -X POST "${{ secrets.AZURE_APP_WEBHOOK_URL }}"
```
[the remark this answers is not in the document]
Comment (emifly) [reply .github/workflows/ci-cd-pipeline.yml]: Thanks, the single quotes seem to work really well!

**Pull request merged**